source/auth/tests/mongodb-oidc.rst

```diff
@@ -39,149 +38,65 @@ is one principal configured.
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a request callback that returns a valid token.
-- Create a client with a url of the form  ``mongodb://test_user1@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Create a client with ``MONGODB_URI_MULTI``, a username of ``test_user1``, and the OIDC request callback.
 - Perform a ``find`` operation that succeeds.
 - Close the client.
 
 Multiple Principal User 2
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
 - Create a request callback that reads in the generated ``test_user2`` token file.
-- Create a client with a url of the form ``mongodb://test_user2@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Create a client with ``MONGODB_URI_MULTI``, a username of ``test_user2``, and the OIDC request callback.
 - Perform a ``find`` operation that succeeds.
 - Close the client.
 
 Multiple Principal No User
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a client with a url of the form ``mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred`` and a valid OIDC request callback.
+- Create a client with ``MONGODB_URI_MULTI``, no username, and the OIDC request callback.
 - Assert that a ``find`` operation fails.
 - Close the client.
 
 Allowed Hosts Blocked
 ~~~~~~~~~~~~~~~~~~~~~
-- Clear the cache.
-- Create a client that uses the OIDC url and a request callback, and an
-  ``ALLOWED_HOSTS`` that is an empty list.
+- Create a default OIDC client, with an ``ALLOWED_HOSTS`` that is an empty list.
 - Assert that a ``find`` operation fails with a client-side error.
 - Close the client.
 - Create a client that uses the url ``mongodb://localhost/?authMechanism=MONGODB-OIDC&ignored=example.com`` a request callback, and an
-  ``ALLOWED_HOSTS`` that contains ["example.com"].
+  ``ALLOWED_HOSTS`` that contains ``["example.com"]``.
 - Assert that a ``find`` operation fails with a client-side error.
 - Close the client.
 
-Lock Avoids Extra Callback Calls
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-- Clear the cache.
-- Create a request callback that returns a token that will expire soon, and
-  a refresh callback.  Ensure that the request callback has a time delay, and
-  that we can record the number of times each callback is called.
-- Spawn two threads or async operations that do the following:
-  - Create a client with the callbacks.
-  - Run a find operation that succeeds.
-  - Close the client.
-  - Create a new client with the callbacks.
-  - Run a find operation that succeeds.
-  - Close the client.
-- Join the two threads or simultaneously call the async operations
-- Ensure that the request callback has been called once, and the refresh
-  callback has been called twice, or that no async function has been
-  entered simultaneously.
-
-AWS Automatic Auth
```

Comment:
Shouldn't we be keeping these tests and changing the URIs from localhost to `MONGODB_URI_SINGLE` and `MONGODB_URI_MULTI` with a `PROVIDER_NAME` of aws?

Reply:
`MONGODB_URI_MULTI` will only be used for human workflows, which will continue to be prose tests only.